### PR TITLE
Feature: grep-able log for emba

### PR DESCRIPTION
**Pull request by p4cx**
Start emba with additional parameter "-g" and it will create a fw_grep_log.log file in your specified log directory. 
There will be every line from the cli in a standardized format:
TYPE;MODULE_NR;SUBMODULE_NR;MESSAGE
e.g.:
MODULE_TITLE;30;0;Version Vulnerability Check
MESSAGE;30;0;Vulnerability database found!
MESSAGE;30;0;Found potential version details in /bin/busybox (-rwxr-xr-x linux linux)
POSITIVE;30;0;Found 26 possible vulnerabilities for busybox [...]

### CODE REVIEW

File: emba.sh [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 # Description:  Main script for load all necessary files and call main function of modules
 
+INVOCATION_PATH="."
```
Comment (m-1-k-3) [inline emba.sh:17]: Please split this pull request into two:
1st - cosmetic updates like this (we can accept this)
2nd - grep output pull request (we should test this)

File: emba.sh [hunk 2]
```diff
@@ -49,6 +50,7 @@ import_module()
 
 main()
 {
+  INVOCATION_PATH="$(dirname "$0")"
```
Comment (m-1-k-3) [inline emba.sh:53]: 1st - cosmetic updates like this (we can accept this)

File: emba.sh [hunk 3]
```diff
@@ -63,15 +65,16 @@ main()
...
-  export LOG_DIR="./logs"
-  export CONFIG_DIR="./config"
-  export EXT_DIR="./external"
-  export HELP_DIR="./helpers"
-  export MOD_DIR="./modules"
+  export LOG_DIR="$INVOCATION_PATH""/logs"
+  export CONFIG_DIR="$INVOCATION_PATH""/config"
+  export EXT_DIR="$INVOCATION_PATH""/external"
+  export HELP_DIR="$INVOCATION_PATH""/helpers"
+  export MOD_DIR="$INVOCATION_PATH""/modules"
```
Comment (m-1-k-3) [inline emba.sh:77]: 1st - cosmetic updates like this (we can accept this)

File: modules/S05_firmware_details.sh [hunk 4]
```diff
@@ -20,7 +20,7 @@
 
 S05_firmware_details()
 {
-  module_log_init "firmware_testing_details"
+  module_log_init "S05_firmware_testing_details"
```
Comment (m-1-k-3) [inline modules/S05_firmware_details.sh:23]: cosmetic update.
Why not using:
module_log_init "${FUNCNAME[0]}"

File: modules/S10_binaries_check.sh [hunk 5]
```diff
@@ -21,7 +21,7 @@
```
Comment (m-1-k-3) [inline modules/S10_binaries_check.sh:21]: all cosmetic updates?

Why not using:
module_log_init "${FUNCNAME[0]}"

File: modules/template_module.sh [hunk 6]
```diff
@@ -31,7 +31,7 @@
 
 empty_module() {
   # Initialize module and creates a log file "empty_module_log.txt" in your log folder
-  module_log_init "empty_module_log"
+  module_log_init "S00_empty_module_log"
```
Comment (m-1-k-3) [inline modules/template_module.sh:34]: cosmetic update.
Why not using:
module_log_init "${FUNCNAME[0]}"

**Comment by m-1-k-3** [comment]
could you please split this into two pull requests? Would be faster to test it ...

File: modules/S05_firmware_details.sh (hunk 4 above)
Comment (p4cx) [reply modules/S05_firmware_details.sh:23]: Why I didn't use it: I am a bad programmer ;) Sounds reasonable and I will implement it and also split this pr in two different pr's.

File: emba.sh (hunk 3 above)
Comment (p4cx) [reply emba.sh:77]: This is not only cosmetic: I needed that for this feature.

File: emba.sh (hunk 2 above)
Comment (p4cx) [reply emba.sh:53]: This is not only cosmetic: I needed that for this feature.

**Comment by p4cx** [comment]
> could you please split this into two pull requests? Would be faster to test it ...

it's done - all cosmetic updates on emba already are in master. It should now be faster to review.

**Review by m-1-k-3 — APPROVED** [review]
this is a great extension!

**Pull request merged**